test/parselink.test.js

```diff
@@ -1,12 +1,19 @@
-import { parseForRollOrDamage, parselink } from '../lib/parselink'
-import { DamageTable } from '../module/damage/damage-tables'
-
-globalThis.GURPS = {}
-globalThis.game = {
-  i18n: {
-    localize: str => str,
-  },
-}
+import { parseForRollOrDamage, parselink } from '@/lib/parselink.js'
+import { DamageTable } from '@/module/damage/damage-tables.js'
+
+beforeAll(() => {
+  /** @type {Game} */
+  globalThis.GURPS = {}
+
+  // HACK: this is used to get around testing in a non-Foundry environment
+  // it should be removed in favour of a proper test environment
```

Comment:
This isn't a hack. It's a necessary part of running a unit test outside of a node container or browser. Unless you know something I don't -- let me know if so. Otherwise remove the comment.

Reply:
When I added this I was thinking about some sort of module I saw a while back that let you simulate the FoundryVTT environment without a full instance of it running, so these global objects would work. However, unless that becomes something more concrete (i.e. I find the thing and decide it's good enough to propose using) , I agree that it's not really a hack. Comment removed in commit.

Reply:
I'll be pedantic here, so forgive me. My definition of unit tests are the tests that can run while we are editing files in the IDE, so that we get feedback instantly when we break something. It would also be beneficial to have integration or component tests, which are run in a production-like environment. The unit tests are almost more important, because the faster we get feedback, the lower the effort to keep the code clean and running.